text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/MiniMessageParser.java

```diff
@@ -120,45 +120,47 @@ private void processTokens(final @NotNull StringBuilder sb, final @NotNull Strin
 
   @NotNull Component parseFormat(final @NotNull String richMessage, final @NotNull ContextImpl context) {
     final PlaceholderResolver combinedResolver = PlaceholderResolver.combining(context.placeholderResolver(), this.placeholderResolver);
-    final Appendable debug = context.debugOutput();
+    final Consumer<String> debug = context.debugOutput();
     if (debug != null) {
-      try {
-        debug.append("Beginning parsing message ").append(richMessage).append('\n');
-      } catch (final IOException ignored) {
-      }
+      debug.accept("Beginning parsing message ");
+      debug.accept(richMessage);
+      debug.accept("\n");
```

Comment:
Should we combine this into one `accept` call instead?

Reply:
ideally we avoid string concat here, since that would produce an unnecessary StringBuilder